tests/functional/settings_clients.js

```diff
@@ -178,9 +181,33 @@ define([
         .end()
 
         // clicking disconnect on the second device should update the list
-        .findByCssSelector('.client:nth-child(2) .client-disconnect')
-          .click()
-        .end()
+        .then(click('.client:nth-child(2) .client-disconnect'))
+
+        // get the modal dialog
+        .then(testElementExists('.intro'))
+        .then(testElementExists('.disabled'))
+
+        // test cancel
+        .then(click('.cancel-disconnect'))
+        .then(FunctionalHelpers.pollUntil(function () {
+          var modalPanel = document.querySelectorAll('#client-disconnect').length;
+
+          return modalPanel === 0 ? true : null;
+        }, [ ], 10000))
+
+        .then(click('.client:nth-child(2) .client-disconnect'))
+        .then(click('select.disconnect-reasons > option[value="lost"]'))
+
+        // wait until button is enabled
```

Comment:
`is enabled` or `has gone away`? Can `waitForDeletedByCssSelector` be used here too?